### ISSUE TITLE
Serialization of proxy types does not encode the proxy type, only the inner type.

### ISSUE THREAD

**Issue by olson-sean-k** [olson-sean-k/decorum#5]
The [implementation of serialization](https://github.com/olson-sean-k/decorum/blob/8fdd4302527cb6e7e470d252c1240ab4aa1ffec7/src/proxy.rs#L1409) with serde is incorrect. Only the raw floating point value is encoded, omitting any information about the originating proxy type. For example, serialization should probably use `serialize_newtype_struct` instead of `into_raw_float` + `serialize`.

Consider using `cfg_attr` with `derive(Deserialize, Serialize)`, though this may not be possible due to the type parameters on `ConstrainedFloat`.

**Comment by olson-sean-k** [comment]
This is fixed by 9e21dee, which uses serde_derive and `cfg_attr` to implement serialization. Note that this results in using an implementation vendored by serde, so this sort of defers on correctness.

**Comment by schungx** [comment]
Is there a way to have the old behavior back?  For example, to serialize/deserialize into raw floats.

When I use `decorum::R32` and serialize to JSON, it constantly comes in as a hash object together with the phantom data field instead of just a number as I need it to be.

**Comment by olson-sean-k** [comment]
Sorry for missing this! I'm actually working on a few tweaks to serialization for [the `0.4.x` series](https://github.com/olson-sean-k/decorum/tree/v0.4.0). I know you commented a long time ago, but just in case this is still relevant to you, I've described this a bit below.

I think it's important for proxy types to be serialized as bona fide types and not raw floating-point primitives. If there's a need to serialized to and from primitive types, then I suggest converting when de/serializing. I've just pushed a change (0611654) that skips the `PhantomData` field, so if serializing proxy types was simply too verbose, then perhaps that will make things more tolerable. That change should make it into the `0.4.0` release.
